src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Location/ShowLocationHistory/ShowLocationHistory.xaml.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright 2019 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+// language governing permissions and limitations under the License.
+
+using Esri.ArcGISRuntime.Geometry;
+using Esri.ArcGISRuntime.Location;
+using Esri.ArcGISRuntime.Mapping;
+using Esri.ArcGISRuntime.Symbology;
+using Esri.ArcGISRuntime.UI;
+using System;
+using System.Windows;
+
+namespace ArcGISRuntime.WPF.Samples.ShowLocationHistory
+{
+    [ArcGISRuntime.Samples.Shared.Attributes.Sample(
+        "Show location history",
+        "Location",
+        "Display your location history on the map.",
+        "")]
+    [ArcGISRuntime.Samples.Shared.Attributes.ClassFile("FakeLocationDataSource.cs")]
+    public partial class ShowLocationHistory
+    {
+        // Track whether location tracking is enabled.
+        private bool _isTrackingEnabled;
+
+        // Location data source provides location data updates.
+        private LocationDataSource _locationDataSource;
+
+        // Graphics overlay to display the location history (points).
+        private GraphicsOverlay _locationHistoryOverlay;
+
+        // Graphics overlay to display the line created by the location points.
+        private GraphicsOverlay _locationHistoryLineOverlay;
+
+        // Polyline builder to more efficiently manage large location history graphic.
+        private PolylineBuilder _polylineBuilder;
+
+        // Track previous location to ensure the route line appears behind the animating location symbol.
+        private MapPoint _lastPosition;
+
+        public ShowLocationHistory()
+        {
+            InitializeComponent();
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            // Create new Map with basemap.
+            Map myMap = new Map(Basemap.CreateDarkGrayCanvasVector());
```

Comment:
Canvas is too dark for the blue line. I would consider changing to the LightGrayCanvasVector or changing the line color.